src/NHSD.BuyingCatalogue.Solutions.Application/Persistence/SupplierReader.cs

```diff
@@ -0,0 +1,23 @@
+using System.Threading;
+using System.Threading.Tasks;
+using NHSD.BuyingCatalogue.Solutions.Application.Domain.Suppliers;
+using NHSD.BuyingCatalogue.Solutions.Contracts.Persistence;
+
+namespace NHSD.BuyingCatalogue.Solutions.Application.Persistence
+{
+    internal sealed class SupplierReader
+    {
+        private readonly ISupplierRepository _supplierRepository;
+
+        public SupplierReader(ISupplierRepository supplierRepository)
+        {
+            _supplierRepository = supplierRepository;
+        }
+
+        public async Task<Supplier> BySolutionIdAsync(string id, CancellationToken cancellationToken)
```

Comment:
Could we change the parameter name to be solutionId? Just to be specific even though the method name suggests it will be.